### ISSUE TITLE
timeout epic

### ISSUE THREAD

**Issue by viperdriver2000** [vogler/free-games-claimer#25]
hi 
i got timeouts.

```
/usr/bin/docker run --rm -p 6080:6080 -p 5900:5900 -v /mnt/nfs/export/docker/free-games-claimer_jackblk/epic-games/data:/fgc/data --name free-games-claimer_jackblk-epic-games free-games-claimer-epic-games
```

```
userAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.83 Safari/537.36
Signed in.
Number of free games: 2
locator.innerText: Timeout 20000ms exceeded.
=========================== logs ===========================
waiting for selector "//button[@data-testid="purchase-cta-button"][not(contains(.,"Loading"))] >> nth=0"
============================================================
    at file:///fgc/epic-games.js:71:124 {
  name: 'TimeoutError'
}
```

any ideas about that?
thanks

**Comment by vogler** [comment]
I have the same issue. Seems to be a race. When stepping through with `PWDEBUG=1 node epic-games` there is no problem.
Strange thing is that it finds the two elements to click on but then the click somehow makes it navigate to a 404 page.

**Comment by vogler** [comment]
Should be fixed for now, but not a great solution.
As the commit says, should wait for some element/attribute/event instead.

**Comment by vogler** [comment]
Apparently it's not time-related after all. Even with `networkidle` and `waitForTimeout(2000)` it sometimes (but rarely) got into this situation.

**Comment by viperdriver2000** [comment]
thanks for your feedback.
im on commit cbe789b08d3cb04e52e9bb975ad79b1a1274675e (HEAD -> main, origin/main, origin/HEAD)

but i get the same error.

```
userAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.83 Safari/537.36
Signed in.
Number of free games: 2
locator.innerText: Timeout 20000ms exceeded.
=========================== logs ===========================
waiting for selector "//button[@data-testid="purchase-cta-button"][not(contains(.,"Loading"))] >> nth=0"
============================================================
    at file:///fgc/epic-games.js:71:124 {
  name: 'TimeoutError'
}
```

**Comment by viperdriver2000** [comment]
thanks for your change.
im on commit 1dc3b3db6af5314b611eb3d9972d83fb451009a0 (HEAD -> main, origin/main, origin/HEAD)

get a new error:
```
userAgent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.83 Safari/537.36
Signed in.
Number of free games: 2
Current free game: ARK: Survival Evolved
Already in library! Nothing to claim.
Current free game: Gloomhaven
Not in library yet! Click GET.
page.waitForSelector: Timeout 20000ms exceeded.
=========================== logs ===========================
waiting for selector "text=Thank you for buying" to be visible
============================================================
    at file:///fgc/epic-games.js:115:20 {
  name: 'TimeoutError'
}
Saved a screenshot of hcaptcha challenge to /fgc/data/screenshots/epic-games/2022-09-23 06.17.11.360.png
Got hcaptcha challenge. To avoid it, get a link from https://www.hcaptcha.com/accessibility
```

**Comment by vogler** [comment]
Did you check if you got a hcaptcha challenge?
I think they got more aggressive. While testing a couple of times in a row it also showed me a captcha and I had to get a fresh IP to avoid it.

**Comment by viperdriver2000** [comment]
okay i see there was a hcaptcha challenge.
okay i will check it next week again ^^

**Comment by vogler** [comment]
Ok, with [69d771b](https://github.com/vogler/free-games-claimer/commit/69d771b38c9aeebb8e4850c6ea983aef535b6fb6) I just had the case where the printed urlSlug was correct, but the click resulted in a 404. I'll change the code to just visit those links then instead of clicking.

**Comment by vogler** [comment]
Should be fixed by [edb90fe](https://github.com/vogler/free-games-claimer/commit/edb90fe9f846fe3e40dd38131d445eadee20a76a). Please reopen if you still encounter the issue after updating.
